vizro-core/docs/pages/user-guides/notification-actions.md

```diff
@@ -0,0 +1,470 @@
+                text: Warning
+                actions:
+                  - type: show_notification
+                    message: Please review your inputs
+                    variant: warning
+              - type: button
+                text: Error
+                actions:
+                  - type: show_notification
+                    message: Something went wrong
+                    variant: error
+              - type: button
+                text: Progress
+                actions:
+                  - type: show_notification
+                    message: Processing...
+                    variant: progress
+                    auto_close: false
+            title: Notification variants
+            layout:
+              type: flex
+              direction: row
+        ```
+
+    === "Result"
+
+        [![NotificationVariants]][notificationvariants]
+
+### Custom title and icon
+
+By default, notifications use the capitalized variant name as the title (e.g., "Info", "Success", "Warning", "Error"). You can customize both the title and icon. The `icon` argument accepts any icon name from the [Google Material Icons library](https://fonts.google.com/icons):
+
+!!! example "Custom title and icon"
+
+    === "app.py"
+
+        ```{.python pycafe-link}
+        import vizro.actions as va
+        import vizro.models as vm
+        from vizro import Vizro
+
+        page = vm.Page(
+            title="Custom notification",
+            components=[
+                vm.Button(
+                    text="Download data",
+                    actions=[
+                        va.show_notification(
+                            title="Download Complete",
+                            message="Your data has been exported successfully",
+                            variant="success",
+                            icon="download",
+                        )
+                    ],
+                ),
+            ],
+        )
+
+        dashboard = vm.Dashboard(pages=[page])
+        Vizro().build(dashboard).run()
+        ```
+
+    === "app.yaml"
+
+        ```yaml
+        # Still requires a .py to add data to the data manager and parse YAML configuration
+        # See yaml_version example
+        pages:
+          - components:
+              - type: button
+                text: Download data
+                actions:
+                  - type: show_notification
+                    title: Download Complete
+                    message: Your data has been exported successfully
+                    variant: success
+                    icon: download
+            title: Custom notification
+        ```
+
+    === "Result"
+
+        [![CustomNotification]][customnotification]
+
+### Auto-close duration
+
+By default, notifications auto-dismiss after 4 seconds (4000 milliseconds). You can customize this duration or disable auto-close entirely. Set `auto_close=False` for important messages that require user acknowledgment.
```

Comment:
Any guidelines on when you should (or should not) require interaction from the user. They can be pretty annoying, maybe link to some accessiblity guidance around why not to overuse dialogs that require dismissal?